src/index.js

```diff
@@ -4,19 +4,19 @@ export const BEGIN = '@@optimist/BEGIN';
 export const COMMIT = '@@optimist/COMMIT';
 export const REVERT = '@@optimist/REVERT';
 
-export const ensureState = state => {
-  if (Map.isMap(state)) {
-    if (List.isList(state.get('history'))) {
-      return state.get('current');
-    }
-  }
-  return state;
-};
+// this marker is used to identify our wrapped state. The marker is designed to prevent false positives
+// that might occur if we checked for the existence of other properties. It is exported for testing.
+export const STATE_MARKER = '__redux-optimistic-ui';
```

Comment:
Yep, after looking at it again I'm going to agree. Your slightly more strict check (all the props + key length) will also help. I'm probably overestimating the chance of a false positive!